packages/web/src/components/usdc-purchase-details-modal/components/PurchaseModalContent.tsx

```diff
@@ -40,25 +51,51 @@ const messages = {
   done: 'Done',
   purchaseDetails: 'Purchase Details',
   track: 'Track',
+  album: 'Album',
   visitTrack: 'Visit Track',
+  visitAlbum: 'Visit Album',
   transaction: 'Explore Transaction'
 }
 
 export const PurchaseModalContent = ({
   purchaseDetails,
   onClose
-}: ContentProps) => {
+}: {
+  purchaseDetails: USDCPurchaseDetails
+  onClose: () => void
+}) => {
+  const goToRoute = useGoToRoute()
+  const { contentType, contentId } = purchaseDetails
+  const isTrack = contentType === USDCContentPurchaseType.TRACK
   const { isEnabled: isPremiumAlbumsEnabled } = useFlag(
     FeatureFlags.PREMIUM_ALBUMS_ENABLED
   )
-  const goToRoute = useGoToRoute()
-  const { data: track } = useGetTrackById({ id: purchaseDetails.contentId })
+  const { data: currentUserId } = useGetCurrentUserId({})
+  const { data: track } = useGetTrackById(
```

Comment:
I'm tempted to suggest that we just split this into a separate component fro each type instead of instantiating hooks that are disabled and sprinkling checks for `track` throughout this component.

Reply:
i am also tempted... but then we will have 4x components: track+sale, track+purchase, album+sale, album+purchase. and the code is almost exactly duplicated unlike `TrackScreenDetailsTile` where the track vs playlist cases have largely diverged.

Reply:
ah just read your comment above, that seems more reasonable.